src/main/java/carRacing/model/RacingRecords.java

```diff
@@ -28,4 +22,11 @@ public List<RacingCar> recordsOfCars(){
         return this.records;
     }
 
+    public List<RacingCar> raceRecord() {
+        for (RacingCar racingCar: this.records) {
+            racingCar.move(moving());
+        }
+        return this.records;
+    }
```

Comment:
moving 메소드가 호출될 때 마다 전략을 인스턴스화하고 있습니다.
전략은 하나이고 이 전략을 모든 자동차에게 전달해보면 어떨까요?? 불필요한 객체 생성을 하지 않아도됩니다.

```suggestion
    public List<RacingCar> raceRecord(RandomMoveStrategy strategy) {
        for (RacingCar racingCar : this.records) {
            racingCar.move(strategy);
        }
        return this.records;
    }
```

이러면 RacingRecords 도 테스트가 가능해지겠네요 👍

Reply:
클래스 내부에서 객체 생성이 이뤄지면 테스트를 쉽게할 수 없습니다.
대표적인 예로 랜덤과 시간인데요. 
랜덤은 이번 코드에서 작성해봤으니 시간을 예로 들어보겠습니다.

수현님이 오늘 하루가 짝수일인지 판단하는 기능을 구현해야 한다고 했을 때 
다음과 같이 구현해볼 수 있습니다.

```java
boolean even()
```

Reply:
클래스 내부에서 객체 생성이 이뤄지면 테스트를 쉽게할 수 없습니다.
대표적인 예로 랜덤과 시간인데요. 
랜덤은 이번 코드에서 작성해봤으니 시간을 예로 들어보겠습니다.

수현님이 오늘 하루가 짝수일인지 판단하는 기능을 구현해야 한다고 했을 때 
다음과 같이 구현해볼 수 있습니다.

```java
public boolean even() {
   // LocalDate.now() 생성 후 날짜가 짝수인지 판단
}
```
만약 여기서 기능을 테스트해보고 싶을 때 어떻게 테스트를 해야할까요?
방법은 여러가지가 있습니다.
- 짝수일에 맞춰서 기능을 돌려본다.
- 시스템 시간을 변경하여 기능을 돌려본다.
- ...
위 방법도 가능하지만 제약이 너무나도 큽니다.

그래서 이런 문제를 해결하기 위해 테스트하기 어려운 값을 바깥에서 주입하는 형태로 변경하는 것입니다.

```java
public boolean even(LocalDate now) {
   // now 가 짝수인지 판단
}
```
이런 형태라면 even 메소드에 짝수일인 LocalDate 객체를 만들어 넣어봄으로써 테스트를 쉽게 작성할 수 있습니다.
감이 좀 잡히실까요?